requests_oauth2client/dpop.py

```diff
@@ -106,22 +132,59 @@ def __init__(
             kwargs=kwargs,
         )
 
+    def _response_hook(self, response: requests.Response, **kwargs: Any) -> requests.Response:
+        """Handles a Resource Server provided DPoP nonce."""
+        if response.status_code == codes.unauthorized and response.headers.get("WWW-Authenticate", "").startswith(
+            "DPoP"
+        ):
+            self.dpop_key.handle_rs_provided_dpop_nonce(response)
+            request = self(response.request.copy())
+            return response.connection.send(request, **kwargs)
```

Comment:
Could probably just call `request.deregister_hook("response", self._response_hook)` here to prevent infinite looping?